tasks/localizationTasks.ts

```diff
@@ -0,0 +1,174 @@
+const locFiles = ['bundle.l10n.%s.json', 'package.nls.%s.json'];
+
+function getGeneratedLocalizationChanges(diffFilesAndDirectories: string[]): string[] {
+    if (diffFilesAndDirectories.length == 0) {
+        return [];
+    }
+
+    const allPossibleLocalizationFileNames = getAllPossibleLocalizationFileNames();
+    const changedLocFilesOrDirectory = [];
+
+    for (const diffFileOrDirectory of diffFilesAndDirectories) {
+        const stat = fs.statSync(diffFileOrDirectory);
+        if (
+            stat.isFile() &&
+            allPossibleLocalizationFileNames.some((locFileName) => path.basename(diffFileOrDirectory) === locFileName)
+        ) {
+            console.log(`${diffFileOrDirectory} is changed.`);
+            changedLocFilesOrDirectory.push(diffFileOrDirectory);
+        }
+
+        if (stat.isDirectory() && diffFileOrDirectory !== 'l10n') {
+            console.log('l10n is changed.');
+            changedLocFilesOrDirectory.push('l10n');
+        }
+    }
+
+    return changedLocFilesOrDirectory;
+}
+
+function getAllPossibleLocalizationFileNames(): string[] {
+    const files = [];
+    for (const lang of localizationLanguages) {
+        for (const file of locFiles) {
+            files.push(util.format(file, lang));
+        }
+    }
+    // English
+    files.push('bundle.l10n.json');
+    return files;
+}
+
+async function git_diff(args: string[]): Promise<string[]> {
+    const result = await git(['diff'].concat(args));
+    // Line ending from the stdout of git is '\n' even on Windows.
+    return result
+        .replaceAll('\n', EOL)
+        .split(EOL)
+        .map((fileName, _) => fileName.trim())
+        .filter((fileName) => fileName.length !== 0);
+}
+
+async function git(args: string[], printCommand = true): Promise<string> {
+    if (printCommand) {
+        console.log(`git ${args.join(' ')}`);
+    }
+
+    const git = spawnSync('git', args);
+    if (git.status != 0) {
+        const err = git.stderr.toString();
+        if (printCommand) {
+            console.log(`Failed to execute git ${args.join(' ')}.`);
+        }
+        throw err;
+    }
+
+    const stdout = git.stdout.toString();
+    if (printCommand) {
+        console.log(stdout);
+    }
+    return stdout;
+}
+
+gulp.task('publish localization content', async () => {
+    const parsedArgs = minimist<Options>(process.argv.slice(2));
+    await git(['add', '-A']);
+    const diffResults = await git_diff(['--name-only', 'HEAD']);
+    if (diffResults.length == 0) {
+        console.log('No git file changes');
+        return;
+    }
+
+    const localizationChanges = getGeneratedLocalizationChanges(diffResults);
+    if (localizationChanges.length == 0) {
+        console.log('No localization file changed');
+        return;
+    }
+    await git(['reset']);
+    const newBranchName = `localization/${parsedArgs.commitSha}`;
+    await git(['config', '--local', 'user.name', parsedArgs.userName]);
```

Comment:
we could potentially make these optional so when you run this locally you don't have to provide them (but we still provide them in CI)